packages/e2e/test/blockfrost/queryTransactions.test.ts

```diff
@@ -0,0 +1,304 @@
+import * as envalid from 'envalid';
+import { Cardano, ChainHistoryProvider } from '@cardano-sdk/core';
+import { chainHistoryProviderFactory } from '../../src/factories';
+import { logger } from '@cardano-sdk/util-dev';
+
+// Verify environment.
+export const env = envalid.cleanEnv(process.env, {
+  CHAIN_HISTORY_PROVIDER: envalid.str(),
+  CHAIN_HISTORY_PROVIDER_PARAMS: envalid.json({ default: {} })
+});
+
+describe('blockfrostChainHistoryProvider', () => {
+  let chainHistoryProvider: ChainHistoryProvider;
+
+  beforeAll(async () => {
+    chainHistoryProvider = await chainHistoryProviderFactory.create(
+      env.CHAIN_HISTORY_PROVIDER,
+      env.CHAIN_HISTORY_PROVIDER_PARAMS,
+      logger
+    );
+  });
+
+  describe('transactionsByHashes', () => {
+    it('parses metadata correctly', async () => {
+      const [tx] = await chainHistoryProvider.transactionsByHashes({
+        ids: [Cardano.TransactionId('5127f8b235111854be744d36703f2098102d2639524035299658e73d683c0dfd')]
+      });
+      expect(tx.auxiliaryData!.blob!.get(1n)).toEqual(
+        new Map<string, string>([
+          ['absolute_slot', '67255007'],
+          ['timestamp', '1722938207']
+        ])
+      );
+    });
+
+    // TODO: waiting for resolution of
+    // https://github.com/blockfrost/openapi/issues/211
+    it.skip('parses collaterals correctly', async () => {
+      const [tx] = await chainHistoryProvider.transactionsByHashes({
+        ids: [Cardano.TransactionId('12fa9af65e21b36ec4dc4cbce478e911d52585adb46f2b4fe3d6563e7ee5a61a')]
+      });
+      expect(tx.body.inputs!.length).toEqual(1);
+      expect(tx.body.outputs!.length).toEqual(1);
+      expect(tx.body.collaterals!.length).toEqual(2);
+      expect(tx.body.collaterals![0]).toEqual({
+        address:
+          'addr_test1qrv0j69s06vd56365fsh5ync44ykaqr4exwf2vt6tuv76' +
+          'lcxy8jmk9rapjmuk2e7cfmshs27r4sx7tk0q3afktjf7j2qvdncx7',
+        index: 1,
+        txId: '2db6592c4782064295295b365c2e8ce84084fa24b1b3f5834f3c6b65268b7878'
+      });
+    });
+    // TODO: waiting for release/deploy in db-sync/blockfrost
+    // https://github.com/input-output-hk/cardano-db-sync/issues/1019
+    it.skip('has collaterals for failed contract', async () => {
+      const [tx] = await chainHistoryProvider.transactionsByHashes({
+        ids: [Cardano.TransactionId('9298f499a4c4aeba53a984cb4df0f9a93b7d158da4c2c2d12a06530841f94cd7')]
+      });
+      expect(tx.body.inputs!.length).toEqual(0);
```

Comment:
Ideally it would provide full transaction body. It might not, but it's not really a useful assertion. This is what we should assert instead:
```suggestion
      expect(tx.inputSource).toBe(InputSource.collaterals);
```

Reply:
@mkazlauskas I changed the tests, but I couldn't find a tx in preprod that meets this. Trying to look for it in the db but I am having problem with my disk space to run preprod on my local ^^

Reply:
Which test was this change request for? If blockfrost returns 0 inputs, then it's a tx that failed phase 2, therefore collateral inputs were consumed (not regular inputs).

Reply:
Yes @mkazlauskas I removed the it since I couldn't find an example in preprod. cfc27a17613dae76022395c4a254c03dbda2efa9

Reply:
I think it's an important test, let's try to find an example. We find it by querying db-sync db

Reply:
LGTM